txtorcon/circuit.py

```diff
@@ -519,15 +600,18 @@ def build_timeout_circuit(tor_state, reactor, path, timeout, using_guards=False)
     """
     timed_circuit = []
     d = tor_state.build_circuit(routers=path, using_guards=using_guards)
+    listener = TimeoutCircuitListener()
 
     def get_circuit(c):
+        c.listen(listener)
         timed_circuit.append(c)
```

Comment:
We should probably un-listen as well (e.g. in the cancel, and also when we're done). Not *strictly* necessary, but good practice.